backend/main.py

```
@@ -274,8 +424,91 @@ def emit(event, data):
+                "student_name": student_name,
+            })
+            emit("nfc:status", {"status": "done"})
+            time.sleep(0.4)
+        except KeyboardInterrupt:
+            break
+        except NoCardException:
+            time.sleep(0.2)
+        except CardConnectionException as e:
+            emit("nfc:status", {"status": "error", "message": _format_nfc_error(e)})
+            time.sleep(1.0)
+        except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)